mesonbuild/backend/xcodebackend.py

```diff
@@ -720,6 +739,47 @@ def generate_pbx_build_style(self, objects_dict: PbxDict) -> None:
             settings_dict.add_item('COPY_PHASE_STRIP', 'NO')
             styledict.add_item('name', f'"{name}"')
 
+    def to_shell_script(self, args: CompilerArgs) -> str:
+        quoted_cmd = []
+        for c in args:
+            quoted_cmd.append(c.replace('"', chr(92) + '"'))
+        return f"\"#!/bin/bash\\n{' '.join(quoted_cmd)}\\n\""
```

Comment:
`/bin/bash` would use the `bash` that's provided by macOS and no other. It may be considered future-unsafe as IIRC Apple plans to drop it anytime soon.
I've applied your changes but replacing `bash` with `sh` as used elsewhere in the backend.